libs/src/api/entityManager.ts

```diff
@@ -0,0 +1,325 @@
+const MAX_PLAYLIST_ID = 2147483647
+
+/*
+  API surface for updated data contract interactions.
+  Provides simplified entity management in a generic fashion
+  Handles metadata + file upload etc. for entities such as Playlist/Track/User
+*/
+export class EntityManager extends Base {
+  /**
+   * Generate random integer between two known values
+   */
+  getRandomInt(min: number, max: number): number {
+    min = Math.ceil(min)
+    max = Math.floor(max)
+    return Math.floor(Math.random() * (max - min) + min)
+  }
+
+  /**
+   * Calculate an unoccupied playlist ID
+   * Maximum value is postgres integer max (2147483647)
+   * Minimum value is artificially set to 400000
+   */
+  async getValidPlaylistId(): Promise<number> {
+    // TODO: Confirm collision of ID with disc prov endpoint to account for hidden / private
+    let playlistId: number = this.getRandomInt(MIN_PLAYLIST_ID, MAX_PLAYLIST_ID)
+    let validIdFound: boolean = false
+    while (!validIdFound) {
+      const resp: any = await this.discoveryProvider.getPlaylists(1, 0, [
+        playlistId
+      ])
+      if (resp.length !== 0) {
+        playlistId = this.getRandomInt(MIN_PLAYLIST_ID, MAX_PLAYLIST_ID)
+      } else {
+        validIdFound = true
+      }
+    }
+    return playlistId
+  }
+
+  /**
+   * Playlist default response values
+   */
+  getDefaultPlaylistReponseValues(): PlaylistOperationResponse {
+    return {
+      blockHash: null,
+      blockNumber: null,
+      playlistId: null,
+      error: null
+    }
+  }
+
+  /**
+   * Create a playlist using updated data contracts flow
+   */
+  async createPlaylist({
+    playlistName,
+    trackIds,
+    description,
+    isAlbum,
+    isPrivate,
+    coverArt,
+    logger = console
+  }: {
+    playlistName: string
+    trackIds: number[]
+    description: string
+    isAlbum: boolean
+    isPrivate: boolean
+    coverArt: string
+    logger: Console
+  }): Promise<PlaylistOperationResponse> {
+    const responseValues: PlaylistOperationResponse =
+      this.getDefaultPlaylistReponseValues()
+    try {
+      const currentUserId: string | null =
+        this.userStateManager.getCurrentUserId()
+      if (!currentUserId) {
+        responseValues.error = 'Missing current user ID'
+        return responseValues
+      }
+      const userId: number = parseInt(currentUserId)
+      const createAction = Action.CREATE
+      const entityType = EntityType.PLAYLIST
+      const entityId = await this.getValidPlaylistId()
+      this.REQUIRES(Services.CREATOR_NODE)
+      const updatedPlaylistImage = await this.creatorNode.uploadImage(
+        coverArt,
+        true // square
+      )
+      const web3 = this.ethWeb3Manager.getWeb3()
+      const currentBlockNumber = await web3.eth.getBlockNumber()
+      const currentBlock = await web3.eth.getBlock(currentBlockNumber)
+      const tracks = trackIds.map(trackId => ({ track: trackId, time: currentBlock.timestamp }))
```

Comment:
since you're doing this here - in the client they may not match - is that ok?

Reply:
in this case, i've observed the client will make a request to DP for the confirmedPlaylist but i can move this to client for consistency